### DISCUSSION TITLE
[Feature Request]Create folder for every file and move all files into those folders. // "Rename should move files to new folders if new filenames contains path separation characters"

### DISCUSSION THREAD

**Discussion by mirddes** [doublecmd/doublecmd#1153] [Suggestions]
https://doublecmd.h1n.ru/viewtopic.php?f=5&t=7117

files/file1.file
files/file2.file
files/file3.file
files/file4.file
files/file5.file

becomes

files/file1/file1.file
files/file2/file2.file
files/file3/file3.file
files/file4/file4.file
files/file5/file5.file

**Comment by WhiteChairFromIkea** [comment]
Indeed, this could be achieved if DC's multirename tool would allow path separation character ("\" in windows OS) to be used in destination filename. What I mean:

1. Open multirename tool, change filenames in such a way, that new filename would contain "\";
2. Click OK, files are renamed with respect to full paths, new destination files are moved to appropriate subfolders, if path contains "\" slaches;
3. Currently, adding slashes can be achieved by "Editor..." -> "Edit filenames" menuitem (edit filenpaths as you want by excel).
4. But having slashes in new filenames currently fails:


![5726i](https://user-images.githubusercontent.com/46539653/138587634-a7f18a8f-dceb-4ef0-81d6-0f6e96317252.png)
![5852O](https://user-images.githubusercontent.com/46539653/138587636-8fbe8d05-17ae-45ad-ae34-4d1e16283d90.png)



I think, this issue should be renamed to "Rename should move files to new folders if new filenames contains path separation characters". (Original feature request is one user specific only, I don't think that the author of DC should implement this for free.)

**Comment by WhiteChairFromIkea** [comment]
Hmmm, this is not possible in freecommander also:    
![1458W](https://user-images.githubusercontent.com/46539653/138587868-d7fb071b-5ee4-4d56-9981-3034f30f3e1b.png)

**Comment by mirddes** [comment]
i want the new folders to be renamed with respect to the files current names.

in my use case maintaining the filenames is important. they need to be nested in folders that share their name.

> "Rename should move files to new folders if new filenames contains path separation characters"

**Comment by alexx2000** [comment]
Theoretically it can be done like this:
![1458W](https://imgur.com/N4KGP1z.png)
But DC does not create new directories in multi-rename tool. So it does not work currently.

**Comment by mirddes** [comment]
this would indeed be ideal.

**Comment by pohlymorph** [comment]
Hi @alexx2000, I would second the feature request of @mirddes : I have a similar use case moving (many) files into subfolders that are selected based on the file's date.

e.g.
[N] --> [Y][M]\\[N]
File1 --> 202501\File1
File2 --> 202409\File1

While this works in TC, which creates the directories as necessary, it does not work for DC when the directories do not exist already.

(Thanks go out to all involved making DC a very very useful tool - I use it on Archlinux...)

**Reply by sagamusix** [reply]
Another +1 from me. I have some huge folders with thousands of files, and based on their starting letter I would like to move them into sub folders. If I only had to do this once, creating the folders beforehand would be a no-brainer, but not if you have to do it multiple times in a row. So automatic folder creating as part of the multi-rename tool would be a very welcome addition.
